src/main/java/uk/gov/hmcts/ccd/auditlog/AuditService.java

```diff
@@ -0,0 +1,61 @@
+package uk.gov.hmcts.ccd.auditlog;
+
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.ccd.auditlog.aop.AuditContext;
+import uk.gov.hmcts.ccd.data.SecurityUtils;
+import uk.gov.hmcts.ccd.data.user.CachedUserRepository;
+import uk.gov.hmcts.ccd.data.user.UserRepository;
+
+import javax.servlet.http.HttpServletRequest;
+import java.time.Clock;
+import java.time.LocalDateTime;
+
+import static java.time.format.DateTimeFormatter.ISO_LOCAL_DATE_TIME;
+
+@Service
+public class AuditService {
+
+    private final Clock clock;
+    private final SecurityUtils securityUtils;
+    private final UserRepository userRepository;
+    private final AuditRepository auditRepository;
+
+    public AuditService(@Qualifier("utcClock") final Clock clock,
+                        @Qualifier(CachedUserRepository.QUALIFIER) final UserRepository userRepository,
+                        @Lazy final SecurityUtils securityUtils, final AuditRepository auditRepository) {
+        this.clock = clock;
```

Comment:
Any particular reason why @Lazy final SecurityUtils securityUtils is lazy. Because it seems that this securityUtils attribute is important for the AuditService functionality.   without securityUtils AuditService will not work.   Maybe it is good if it always present.   If securityUtils were inside an if statement yes .

Reply:
Can't work without that - it's not optional. 
SecurityUtils referring AuthTokenGenerator which is loading lazily at the moment.

Reply:
Fair enough . It means that SecurityUtils is liked to Lazy initialization due to it is using AuthTokenGenerator with lazy loading.  Perfect : 👌